sys/include/net/gnrc/netdev.h

```diff
@@ -66,6 +66,27 @@ extern "C" {
  */
 #define GNRC_NETDEV_MAC_INFO_RX_STARTED         (0x0004U)
 
+#ifdef MODULE_GNRC_MAC
+/**
+ * @brief   Flag to track if a device supports on-chip CSMA functionality
+ *
+ * This flag should be (is expected) set only once at the initialization of the
+ * device, or at the initialization of the MAC protocol, according to the fact that
+ * whether the device supports on-chip CSMA. If yes, the related flag bit should
+ * be set, otherwise, should be cleared.
+ */
+#define GNRC_NETDEV_MAC_INFO_ONCHIP_CSMA        (0x0100U)
```

Comment:
This is already handled via netopt by csma_sender_csma_ca_send. If the check in csma_sender is not correct, then it should be refactored instead of adding another layer of checks on top.